pymc/sampling/jax.py

```diff
@@ -289,25 +289,22 @@ def _update_coords_and_dims(
         dims.update(idata_kwargs.pop("dims"))
 
 
-@partial(jax.jit, static_argnums=(2, 3, 4, 5, 6))
 def _blackjax_inference_loop(
-    seed,
-    init_position,
-    logprob_fn,
-    draws,
-    tune,
-    target_accept,
-    algorithm=None,
+    seed, init_position, logprob_fn, draws, tune, target_accept, **adaptation_kwargs
 ):
     import blackjax
 
-    if algorithm is None:
+    algorithm_name = adaptation_kwargs.pop("algorithm", "nuts")
+    if algorithm_name == "nuts":
         algorithm = blackjax.nuts
+    else:
```

Comment:
```suggestion
    elif algorithm_name == "hmc":
```

And raise in the else?

Reply:
Done, thanks